src/labone/nodetree/node.py

```diff
@@ -235,13 +260,9 @@ def construct_nodetree(
         Returns:
             Root-node of the tree.
         """
-        has_common_prefix = len(self._partially_explored_structure.keys()) == 1
-
-        if not hide_kernel_prefix or not has_common_prefix:
-            return self.path_segments_to_node(())
-
-        common_prefix = next(iter(self._partially_explored_structure.keys()))
-        return self.path_segments_to_node((common_prefix,))
+        return self.path_segments_to_node(
+            self.get_root_prefix(hide_kernel_prefix=hide_kernel_prefix)
+        )
```

Comment:
how about we store the root prefix as a member so it's only calculated once (in `construct_nodetree`) and then accessed any time we need it (e.g. in the root property)

Reply:
This would make absolutely sense. Either this, or storing hide_kernel_prefix. But then we would create a new field outside of __init__, which is bad style. So let's pass hide_kernel_prefix straight to the constructor. But this will change the signature and break the instrument, dataserver, half of the tests and the already written 'customer-code'. Is it worth it?